src/StaffHub/Components/Pages/AdminPanel.razor

```diff
@@ -244,33 +244,35 @@
 		}
 	}
 
-	private void CreateUser()
+	private async Task CreateUser()
 	{
-		teams = _unitOfWork.Team.GetAll();
+		bool teamExists = _unitOfWork.Team.GetAll().Any(x => x.Name.ToLower() == team.ToLower());
+		bool mailExists = _unitOfWork.User.GetAll().Any(x => x.Email == email);
 
-		var newTeam = new Team
+		if (mailExists)
 		{
-			Name = team
-		};
-
-		int s = newTeam.Id;
-
-		bool teamExists = false;
-		foreach (var t in teams)
-		{
-			if (team.ToLower() == t.Name.ToLower())
-			{
-				teamId = t.Id;
-				teamExists = true;
-				break;
-			}
+			CloseModal();
+			return;
 		}
 
 		if (!teamExists)
 		{
-			teamId = newTeam.Id;
+			var newTeam = new Team
+				{
+					Name = team.ToLower()
+				};
 			_unitOfWork.Team.Add(newTeam);
 			_unitOfWork.Save();
+			teamId = newTeam.Id;
+		}
+		else
+		{
+			teams = _unitOfWork.Team.GetAll();
+			foreach (var t in teams)
+			{
+				if (team.ToLower() == t.Name.ToLower())
+					teamId = t.Id;
+			}
```

Comment:
This code can be simplified:
~~~c#
var team = _unitOfWork.Team.GetAll().FirstOrDefault(x => x.Name == team.ToLower());
if (team == null) {
  team = new Team() {
    name = team.ToLower(),
  }
}
//continue using the new team.
~~~